### PR TITLE
Microsoft.IISManagerRemoteAdministration version 1.2

### PR DESCRIPTION
**Pull request by Megasware128**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/15446)

### CODE REVIEW

**Comment by Megasware128** [comment]
I've added IIS-ManagementConsole as a Windows Feature dependency

**Comment by Megasware128** [comment]
I've added IIS-ManagementConsole as a Windows Feature dependency

**Comment by Megasware128** [comment]
Accidentally deleted my branch

**Comment by Megasware128** [comment]
Like I mentioned in the previous comments. This PR is waiting for the dependencies feature and I accidentally deleted my branch which resulted in a close of this PR

**Review by ItzLevvie — APPROVED** [review]

**Comment by Megasware128** [comment]
@stephengillie any idea what's going wrong? I don't have access to Windows Sandbox at the moment (new laptop with Windows 11 Home 😞) and I'm not able to interpretate it from the Azure Pipelines logs.

**Comment by stephengillie** [comment]
Automatic Validation ended with:
>

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 1:40:23 PM: Product: Internet Information Services (IIS) 7+ Manager -- The IIS Management Console is not installed, but it is required for managing remote IIS Servers. Please install the IIS Management Console before installing the remote management support by opening the 'Control Panel->Programs->Turn Windows Features On or Off' and selecting the IIS Management Console in the Ineternet Information Services feature.

(Automated response - build 485.) Manual Validation ended with:
> 1:40:23 PM: Product: Internet Information Services (IIS) 7+ Manager -- The IIS Management Console is not installed, but it is required for managing remote IIS Servers. Please install the IIS Management Console before installing the remote management support by opening the 'Control Panel->Programs->Turn Windows Features On or Off' and selecting the IIS Management Console in the Ineternet Information Services feature.

**Comment by stephengillie** [comment]
Hi @Megasware128,

I believe this is still waiting on WindowsFeature support. This was run again because I re-ran our 25 oldest PRs, which happened to include this one.

**Comment by Megasware128** [comment]
This PR has to be on hold until Windows Features become available

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by Trenly** [comment]
[Policy] Windows Features

**Comment by Megasware128** [comment]
@stephengillie I see 1.7 finally has support for Windows Features. Any suggestions to make this succeed?

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> 2024-03-01T21:59:15.0270954Z ##[error] One or more errors occurred. (Manifest: The manifest version is not supported. Supported versions: 1.0.0,1.1.0,1.2.0,1.4.0,1.5.0,1.6.0.) Manifest: The manifest version is not supported. Supported versions: 1.0.0,1.1.0,1.2.0,1.4.0,1.5.0,1.6.0.

(Automated response - build 870.)

File: manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  InstallerSha256: 1ACC8864E0967D67FBBA1C3153205EF4F8B20DC609C01FDAD35A86E606375FC6
+  ProductCode: '{2349E6AA-CFCA-4D17-B633-3ECDA92E38CD}'
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: msi
+  InstallerUrl: https://download.microsoft.com/download/2/4/3/24374C5F-95A3-41D5-B1DF-34D98FF610A3/inetmgr_x86_en-US.msi
+  InstallerSha256: E4EB41CC371F63FAAF9FBDBAD1D8F78FCAB8743E34658D4C81C05441BD1D9E52
+  ProductCode: '{2349E6AA-CFCA-4D17-B633-3ECDA92E38CD}'
+ManifestType: installer
+ManifestVersion: 1.7.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.installer.yaml:22]:
```suggestion
ManifestVersion: 1.6.0
```

File: manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Created using wingetcreate
+
+PackageIdentifier: Microsoft.IISManagerRemoteAdministration
+PackageVersion: 1.2
+DefaultLocale: en-US
+ManifestType: version
+ManifestVersion: 1.7.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.yaml:7]:
```suggestion
ManifestVersion: 1.6.0
```

File: manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+License: Copyright (c) Microsoft Corporation. All rights reserved.
+ShortDescription: IIS Manager for Remote Administration is an administration tool that provides end users and administrators with the ability to remotely manage IIS servers of version 7.0 and above.
+Tags:
+- remote
+- remotemachine
+- remote machine
+- iis
+- windows server
+ManifestType: defaultLocale
+ManifestVersion: 1.7.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/IISManagerRemoteAdministration/1.2/Microsoft.IISManagerRemoteAdministration.locale.en-US.yaml:17]:
```suggestion
ManifestVersion: 1.6.0
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @Megasware128,

We might not be supporting this manifest version quite yet.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 
> 2024-03-13 09:33:29.464 [CORE] Failed to enable Windows Feature [IIS-ManagementConsole] with exit code: 50

(Automated response - build 870.)

**Comment by Megasware128** [comment]
@stephengillie it should work now.

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 
> 2024-03-26 08:52:16.471 [CORE] Failed to enable Windows Feature [IIS-WebServerRole] with exit code: 14081 
> 2024-03-26 08:52:16.472 [CLI ] Terminating context: 0x8a150110 at C:\__w\1\s\external\pkg\src\AppInstallerCLICore\Workflows\DependenciesFlow.cpp:d4 

(Automated response - build 874.)

**Comment by Megasware128** [comment]
Hi @stephengillie, locally on a clean VM it works for me. I'm not really sure how to solve this for Azure Pipelines. Any suggestions?

**Comment by stephengillie** [comment]
I'm not sure either. I see the same error in my VMs, but this might be due to an out-of-date package manager version there.

**Comment by denelon** [comment]
@Megasware128 did you test this on machines with the Windows Optional Feature(s) enabled or disabled? We're still working through the fun with the infrastructure to properly validate Windows Feature enablement. The reboot scenarios are less than cooperative. 😁

**Comment by Megasware128** [comment]
I used the dev machine template from Hyper-V. Last time I checked none of the IIS components were enabled by default.

**Comment by denelon** [comment]
That makes sense. The goal I believe would be to add those Windows Feature dependencies to the manifest here so WinGet can enable the Windows features.

**Comment by Megasware128** [comment]
@denelon whoops. I meant disabled. The IIS features we're NOT enabled. I mistyped 😅

**Comment by Megasware128** [comment]
Hi @denelon, I've done some testing. While it does work on Windows 11 it doesn't work in Windows Server. I'm getting the error: `This product is not required in Windows Server 2008 or Windows Server 2008 R2 or newer. Please install the IIS Management Console by opening the Server Manager and selecting the IIS Management Console in the Role Services for the Web Server Role.`. May this be the issue with Azure Pipelines?

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by denelon** [comment]
@stephengillie if you can run manual validation on this one on Windows 11 desktop, we can go ahead and approve. I've got a longer discussion with some internal teams to address the impacts on Windows Server 2025.

**Comment by denelon** [comment]
@Megasware128 can you update this to the 1.9 schema?

**Comment by Megasware128** [comment]
Sure, I'll try tomorrow

**Comment by stephengillie** [comment]
The install completed successfully in a Windows 11 VM. But I'm not sure where to launch the application, and it doesn't show up in Apps Settings. Is this expected? Where should I be checking for the application?

image

**Comment by Megasware128** [comment]
@stephengillie the fact you see a connections tab with a start page and the ability to add multiple servers means it's working

**Review by stephengillie — APPROVED** [review]

**Comment by stephengillie** [comment]
Note for posterity: This worked on Windows 11 but didn't work on Windows 10.

**Comment by Megasware128** [comment]
I didn't update the manifest to 1.9 yet. Is that a problem?

**Comment by denelon** [comment]
@Megasware128 this got in as an exception. I would appreciate a version bump in the schema if you don't mind, but it isn't strictly necessary.

**Pull request merged**